src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs

```diff
@@ -102,7 +102,7 @@ private static bool IsEnumerableCandidate(DkmClrValue value)
         {
             Debug.Assert(!value.IsError());
 
-            if (value.IsNull)
+            if (value.IsNull || value.HasExceptionThrown())
             {
```

Comment:
This scenario is covered by existing test:  GetEnumerableException

(it just wasn't demonstrating the issue because our test mock didn't throw an Exception)